### PR TITLE
fix: ResizeLV does not check thinpool overprovisioned size

### PR DESCRIPTION
**Pull request by nbalacha**
The ResizeLV function did not use the available overprovisioned
capacity in the size checks for thin-provisioned devices. This caused
resizing a thin LV to fail when the VG did not have sufficient
space but the overprovisioned capacity did.

Signed-off-by: N Balachandran <nibalach@redhat.com>

### CODE REVIEW

File: lvmd/lvservice.go [hunk 1]
```diff
@@ -306,14 +306,39 @@ func (s *lvService) ResizeLV(_ context.Context, req *proto.ResizeLVRequest) (*pr
...
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		tpu, err := pool.Free()
+		if err != nil {
+			log.Error("failed to get free bytes", map[string]interface{}{
+				log.FnError: err,
+			})
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		free = uint64(math.Floor(dc.ThinPoolConfig.OverprovisionRatio*float64(tpu.SizeBytes))) - tpu.VirtualBytes
```
Comment (nbalacha) [inline lvmd/lvservice.go:335]: I would like to move this into a separate function as it is being called multiple times. What is the best file/pkg for this?

Comment (satoru-takeuchi) [reply lvmd/lvservice.go:335]: It's OK to place the new function in `lvmd/lvservice.go` since this code is only called from this file.

**Comment by satoru-takeuchi** [comment]
@nbalacha I'll approve this PR after you add a new function described in the above comment.

File: lvmd/lvservice.go (hunk 1 above)
Comment (nbalacha) [reply lvmd/lvservice.go:335]: For some reason the tests fail when I move this to a function. Looking into it further but if I cannot find out why, can we continue with this PR for now?

Comment (satoru-takeuchi) [reply lvmd/lvservice.go:335]: OK, no problem.

**Review by satoru-takeuchi — APPROVED** [review]

File: lvmd/lvservice.go (hunk 1 above)
Comment (leelavg) [reply lvmd/lvservice.go:335]: - iirc, this is used in vgservice.go as well.
- earlier the calculation used was plannrd to be part of thinpool struct
- after a comment that command.go should only contain struct info the idea was dropped

File: lvmd/lvservice.go [hunk 2]
```diff
@@ -306,14 +306,39 @@ func (s *lvService) ResizeLV(_ context.Context, req *proto.ResizeLVRequest) (*pr
...
-		})
-		return nil, status.Error(codes.Internal, err.Error())
+	free := uint64(0)
+	var pool *command.ThinPool
+	switch dc.Type {
+	case TypeThick:
+		free, err = vg.Free()
+		if err != nil {
+			log.Error("failed to get free bytes", map[string]interface{}{
+				log.FnError: err,
```
Comment (daichimukai) [inline lvmd/lvservice.go:316]: Is `"name": req.GetName()` intentionally dropped?

Comment (nbalacha) [reply lvmd/lvservice.go:316]: I copied the code block from https://github.com/topolvm/topolvm/blob/5df9e7cede433d41c9c77827b0f26dacfa8a2a59/lvmd/lvservice.go#L52

I can add the "name" field if required.

Comment (daichimukai) [reply lvmd/lvservice.go:316]: Understood. Thanks.

**Review by daichimukai — APPROVED** [review]

**Pull request merged**